cascada/primitives/rectangle.py

```diff
@@ -0,0 +1,219 @@
+"""RECTANGLE cipher."""
+from cascada.bitvector.core import Constant
+from cascada.bitvector.operation import RotateLeft
+
+from cascada.bitvector.ssa import RoundBasedFunction
+from cascada.primitives.blockcipher import Encryption, Cipher
+
+
+def sub_column(x):
```

Comment:
For a 4-bit S-box, I would recommend to implement the S-box with a `WDTModel` (see `skinny64` for an example). Otherwise there are too many dependent propagations within the S-box and too many invalid characteristics are found (I did a quick search and this was the case). Can you implement sub_column as in `skinny64` ?

Reply:
Yes, it seems to work much better now (weights are lower and valid). I'm not sure it is all correct and optimal (how bits are rearranged for SboxLut).

NOEKEON works quite well as is (at least weights), but I will try if it works better with WDTModel. It seems simpler S-boxes work better.